src/app/project-notification/documents/upload/upload.component.ts

```diff
@@ -0,0 +1,180 @@
+import { Component, OnInit, ChangeDetectorRef, OnDestroy } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
+import { NgbDateStruct } from '@ng-bootstrap/ng-bootstrap';
+import { Router } from '@angular/router';
+import { Subject, of, forkJoin } from 'rxjs';
+import { MatSnackBar } from '@angular/material';
+import { DocumentService } from 'app/services/document.service';
+import { StorageService } from 'app/services/storage.service';
+
+import { Document } from 'app/models/document';
+
+@Component({
+  selector: 'app-upload',
+  templateUrl: './upload.component.html',
+  styleUrls: ['./upload.component.scss']
+})
+export class UploadComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public currentProject;
+  public projectFiles: Array<File> = [];
+  public documents: Document[] = [];
+  public datePosted: NgbDateStruct = null;
+  public dateUploaded: NgbDateStruct = null;
+  public myForm: FormGroup;
+  public loading = true;
+  public docNameInvalid = false;
+  public legislationYear = '2018';
+  public publishDoc = false;
+
+  constructor(
+    private router: Router,
+    private _changeDetectionRef: ChangeDetectorRef,
+    private storageService: StorageService,
+    private documentService: DocumentService,
+    private snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.currentProject = this.storageService.state.currentProject.data;
+    this.buildForm();
+  }
+
+  buildForm() {
+    this.myForm = new FormGroup({
+      'displayName': new FormControl()
+    });
+    this.loading = false;
+  }
+
+  populateForm() {
+    this.loading = false;
+  }
+
+  register(myForm: FormGroup) {
+    console.log('Successful registration');
+    console.log(myForm);
+  }
+
+  public uploadAndPublish() {
+    this.publishDoc = true;
+    this.uploadDocuments();
+  }
+
+  public uploadDocuments() {
+    this.loading = true;
+
+    // go through and upload one at a time.
+    let observables = [];
+
+    // NB: If multi upload, then switch to use documentFileName as displayName
+
+    this.documents.map(doc => {
+      const formData = new FormData();
+      formData.append('upfile', doc.upfile);
+      formData.append('project', this.currentProject._id);
+      formData.append('documentFileName', doc.documentFileName);
+      formData.append('documentSource', 'PROJECT-NOTIFICATION');
+      formData.append('displayName', this.documents.length > 1 ? doc.documentFileName : this.myForm.value.displayName);
+      formData.append('dateUploaded', new Date().toISOString());
+      formData.append('datePosted', new Date().toISOString());
```

Comment:
I think we need to append legislationYear here, as well. Document uploads are failing for me due to an api validation error on that field.

Reply:
I've added back all the default formData fields, most set to null as we don't need them for project notifications.